patroni/config.py

```diff
@@ -151,10 +159,87 @@ def _safe_copy_dynamic_configuration(self, dynamic_configuration):
                         config['postgresql'][name].update(self._process_postgresql_parameters(value))
                     elif name not in ('connect_address', 'listen', 'data_dir', 'pgpass', 'authentication'):
                         config['postgresql'][name] = deepcopy(value)
-            elif name in config:
+            elif name in config:  # only variables present in __DEFAULT_CONFIG allowed to be overriden from DCS
                 config[name] = int(value)
         return config
 
+    @staticmethod
+    def _build_environment_configuration():
+        ret = defaultdict(dict)
+
+        def _popenv(name):
+            return os.environ.pop('PATRONI_' + name.upper(), None)
```

Comment:
This will break with ``PATRONI_admin_PASSWORD``, as it will change this into ``PATRONI_ADMIN_PASSWORD`` right?